storm-core/src/clj/backtype/storm/util.clj

```diff
@@ -1062,6 +1062,22 @@
 (defn hashmap-to-persistent [^HashMap m]
   (zipmap (.keySet m) (.values m)))
 
+(defn retry-on-exception
+  "Retries specific function on exception based on retries count"
+  [tries task-description f & args]
+  (let [res (try {:value (apply f args)}
+              (catch Exception e
+                (if (= 0 tries)
+                  (throw e)
+                  {:exception e})))]
```

Comment:
Probably should check that `tries` is not `pos?`, in case of crazy values of `tries`.